fastai/tabular/data.py

```diff
@@ -12,22 +12,39 @@ class TabularDataLoaders(DataLoaders):
     "Basic wrapper around several `DataLoader`s with factory methods for tabular data"
     @classmethod
     @delegates(Tabular.dataloaders, but=["dl_type", "dl_kwargs"])
-    def from_df(cls, df, path='.', procs=None, cat_names=None, cont_names=None, y_names=None, y_block=None,
-                valid_idx=None, **kwargs):
-        "Create from `df` in `path` using `procs`"
+    def from_df(cls,
+                df:pd.DataFrame, # A Dataframe object
+                path:str='.', # Location of `df` (defaults to current working directory otherwise)
+                procs:TabularProc=None, # List of `TabularProc`s (non-lazy `Transform`s) applied on object inplace
+                cat_names:list=None, # Column names pertaining to Categorical variables
+                cont_names:list=None, # Column names pertaining to Continuous variables
+                y_names:list=None, # Names of the Dependent variables
+                y_block:TransformBlock=None, # `TransformBlock` to use for the Target
+                valid_idx:list=None, # List of indices to use for the Validation set (defaults to a random split otherwise)
+                **kwargs):
+        "Create `DataLoaders` from `df` in `path` using `procs`"
         if cat_names is None: cat_names = []
         if cont_names is None: cont_names = list(set(df)-set(L(cat_names))-set(L(y_names)))
         splits = RandomSplitter()(df) if valid_idx is None else IndexSplitter(valid_idx)(df)
         to = TabularPandas(df, procs, cat_names, cont_names, y_names, splits=splits, y_block=y_block)
         return to.dataloaders(path=path, **kwargs)
 
     @classmethod
-    def from_csv(cls, csv, skipinitialspace=True, **kwargs):
-        "Create from `csv` file in `path` using `procs`"
+    def from_csv(cls,
+                 csv:str, # Location of a `csv` file
```

Comment:
```suggestion
                 csv:(str,Path,io.BufferedReader), # `csv` file to be ingested
```
Not sure I love my wording here.  Just wanted to mostly show that it can also handle Path and even a BufferedReader since all this does is process pd.read_csv

Reply:
That makes sense, 
Is it fine to take over the type-hint's and comments directly from pandas for this, which is as follows -->
 ```suggestion
                 csv:FilePathOrBuffer, # str, path object or file-like object
```
which satisfies all the required suggestions 😉

Reply:
If i use `FilePathOrBuffer`, it was throwing error, so i have used your suggestion and updated as follows ->
`csv:(str,Path,io.BufferedReader), # str, path object or file-like object`